File: src/main/java/com/example/demo/api/ErrorResponse.java
package com.example.demo.api;

import java.time.LocalDateTime;

public record ErrorResponse(
    String errorCode,
    String message,
    LocalDateTime timestamp
) {

	public static ErrorResponse from(ErrorMessage errorMessage) {
        return new ErrorResponse(
            errorMessage.getCode(),
            errorMessage.getMessage(),
            LocalDateTime.now()
        );
    }
}

File: src/main/kotlin/com/example/demo/api/ErrorResponse.kt
package com.example.demo.api

import java.time.LocalDateTime

data class ErrorResponse(
    val errorCode: String,
    val message: String,
    val timestamp: LocalDateTime
) {
    companion object {
        fun from(errorMessage: ErrorMessage): ErrorResponse {
            return ErrorResponse(
                errorCode = errorMessage.code,
                message = errorMessage.message,
                timestamp = LocalDateTime.now()
            )
        }
    }
}
